test/Dialect/Ibis/round-trip.mlir

```diff
@@ -0,0 +1,56 @@
+// RUN: circt-opt %s | circt-opt | FileCheck %s
+
+// CHECK-LABEL:  ibis.class @A {
+// CHECK-NEXT:    ibis.port.input @A_in : i1
+// CHECK-NEXT:    ibis.port.output @A_out : i1
+// CHECK-NEXT:  }
+
+// CHECK-LABEL:  ibis.class @C {
+// CHECK-NEXT:    ibis.port.input @C_in : i1
+// CHECK-NEXT:    ibis.port.output @C_out : i1
+// CHECK-NEXT:    ibis.instance @My_A, @A
+// CHECK-NEXT:    %true = hw.constant true
+// CHECK-NEXT:    ibis.port.write @C_in(%true) : i1
+// CHECK-NEXT:    %0 = ibis.port.read @C_out : i1
+// CHECK-NEXT:    ibis.instance.write @My_A::@A_in(%0) : i1
+// CHECK-NEXT:    %1 = ibis.instance.read @My_A::@A_out : i1
+// CHECK-NEXT:    ibis.container @D {
+// CHECK-NEXT:      %true_0 = hw.constant true
+// CHECK-NEXT:      ibis.port.write @C_in(%true_0) : i1
+// CHECK-NEXT:      %2 = ibis.port.read @C_out : i1
+// CHECK-NEXT:      ibis.instance.write @My_A::@A_in(%2) : i1
+// CHECK-NEXT:      %3 = ibis.instance.read @My_A::@A_out : i1
+// CHECK-NEXT:    }
+// CHECK-NEXT:  }
+
+ibis.class @A {
+  ibis.port.input @A_in : i1
+  ibis.port.output @A_out : i1
+}
+
+ibis.class @C {
+  ibis.port.input @C_in : i1
+  ibis.port.output @C_out : i1
+
+  ibis.instance @My_A, @A
```

Comment:
Let's stick to the camelCase for instance names and CamelCase for class name convention. (I'm a hypocrite here since I never stick to any conventions, but do as I say not as I do! ;) )